sql/rowexec/show_iters.go

```diff
@@ -260,33 +260,61 @@ func (i *showIndexesIter) Next(ctx *sql.Context) (sql.Row, error) {
 	), nil
 }
 
-func isFirstColInUniqueKey(s *plan.ShowColumns, col *sql.Column, table sql.Table) bool {
+// isPriCol checks if this column is the first column in a unique index
```

Comment:
mysql docs say "the column is a PRIMARY KEY or is one of the columns in a multiple-column PRIMARY KEY."

Reply:
i would have expected this to lookup "PRIMARY" and iterate for comparison

Reply:
Later on in the docs MySQL says
> A UNIQUE index may be displayed as PRI if it cannot contain NULL values and there is no PRIMARY KEY in the table. A UNIQUE index may display as MUL if several columns form a composite UNIQUE index; although the combination of the columns is unique, each column can still hold multiple occurrences of a given value.

[Source](https://dev.mysql.com/doc/refman/8.3/en/show-columns.html#:~:text=A%20UNIQUE%20index%20may%20be%20displayed%20as%20PRI%20if%20it%20cannot%20contain%20NULL%20values%20and%20there%20is%20no%20PRIMARY%20KEY%20in%20the%20table.)

I have no idea why they chose to do this, since the column is still not part of the primary key.

Reply:
hmm, maybe combine the specs into docstrings? the non-nullable part seems important. Would the code be more self-documenting if we stored PRIMARY separately? or different code path if PRIMARY exists?